tests/e2e/functional_test.go

```diff
@@ -353,7 +354,7 @@ var _ = Describe("Functional e2e", Serial, func() {
 		document("Verifying Pipeline got updated")
 
 		// get Pipeline to check that spec has been updated to correct spec
-		verifyPipelineSpec(Namespace, pipelineName, func(retrievedPipelineSpec numaflowv1.PipelineSpec) bool {
+		verifyPipelineSpec(Namespace, getPipelineName(Namespace, pipelineRolloutName), func(retrievedPipelineSpec numaflowv1.PipelineSpec) bool {
```

Comment:
Actually, what I'm thinking we may want to do is query for the "promoted" child inside the `Eventually` calls in the individual functions like `verifyPipelineSpec()`. Ideally, we should be able to call something like `verifyPipelineSpec()` and within the call the "promoted" pipeline should be able to change from `pipeline-0` to `pipeline-1`. `pipeline-0` does not have the spec we want, but `pipeline-1` does.

Reply:
Since this would be a major change, maybe create a second PR and just leave this one up in Draft?

Reply:
Got it, so we just want to remove the pipelineName parameter in the individual functions. I can open a new PR to work on that approach.